### PR TITLE
[wasm] Don't use `_WasmDevel=true` when publishing

### LINKED ISSUE #87414: WasmTemplateTests.ConsolePublishAndRun fails with exit code 137
## Build Information
Build: https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=301956
Build error leg or test failing: Wasm.Build.Tests.WasmTemplateTests.ConsolePublishAndRun
Pull request: https://github.com/dotnet/runtime/pull/87156

Failures in Build browser-wasm linux Release WasmBuildTests, in WasmTemplateTests:

```
Expected 42 exit code but got 137
```
```
 Stack Trace:
        /_/src/mono/wasm/Wasm.Build.Tests/Common/CommandResult.cs(44,0): at Wasm.Build.Tests.CommandResult.EnsureExitCode(Int32 expectedExitCode, String messagePrefix, Boolean suppressOutput)
        /_/src/mono/wasm/Wasm.Build.Tests/WasmTemplateTests.cs(409,0): at Wasm.Build.Tests.WasmTemplateTests.ConsolePublishAndRun(String config, Boolean aot, Boolean relinking)
           at InvokeStub_WasmTemplateTests.ConsolePublishAndRun(Object, Object, IntPtr*)
           at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
```

## Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssueJsonStepByStep.md#how-to-create-a-known-issue-step-by-step).

```json
{
  "ErrorMessage": "mono/wasm/Wasm.Build.Tests/WasmTemplateTests.cs(409,0): at Wasm.Build.Tests.WasmTemplateTests.ConsolePublishAndRun",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

### Report

|Build|Definition|Test|Pull Request|
|---|---|---|---|
|[307611](https://dev.azure.com/dnceng-public/public/_build/results?buildId=307611)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=307611&view=ms.vss-test-web.build-test-results-tab&runId=6258620&resultId=100105)|dotnet/runtime#87570|
|[307434](https://dev.azure.com/dnceng-public/public/_build/results?buildId=307434)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=307434&view=ms.vss-test-web.build-test-results-tab&runId=6255376&resultId=100105)||
|[307389](https://dev.azure.com/dnceng-public/public/_build/results?buildId=307389)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=307389&view=ms.vss-test-web.build-test-results-tab&runId=6254486&resultId=100105)|dotnet/runtime#87575|
|[307377](https://dev.azure.com/dnceng-public/public/_build/results?buildId=307377)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=307377&view=ms.vss-test-web.build-test-results-tab&runId=6254374&resultId=100105)|dotnet/runtime#87285|
|[307369](https://dev.azure.com/dnceng-public/public/_build/results?buildId=307369)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=307369&view=ms.vss-test-web.build-test-results-tab&runId=6254196&resultId=100105)|dotnet/runtime#87573|
|[307349](https://dev.azure.com/dnceng-public/public/_build/results?buildId=307349)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=307349&view=ms.vss-test-web.build-test-results-tab&runId=6253694&resultId=100105)|dotnet/runtime#87570|
|[307341](https://dev.azure.com/dnceng-public/public/_build/results?buildId=307341)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=307341&view=ms.vss-test-web.build-test-results-tab&runId=6253586&resultId=100105)|dotnet/runtime#87566|
|[307309](https://dev.azure.com/dnceng-public/public/_build/results?buildId=307309)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=307309&view=ms.vss-test-web.build-test-results-tab&runId=6253132&resultId=100105)|dotnet/runtime#87549|
|[307283](https://dev.azure.com/dnceng-public/public/_build/results?buildId=307283)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=307283&view=ms.vss-test-web.build-test-results-tab&runId=6252804&resultId=100105)|dotnet/runtime#87567|
|[307101](https://dev.azure.com/dnceng-public/public/_build/results?buildId=307101)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=307101&view=ms.vss-test-web.build-test-results-tab&runId=6248808&resultId=100105)|dotnet/runtime#87272|
|[307035](https://dev.azure.com/dnceng-public/public/_build/results?buildId=307035)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=307035&view=ms.vss-test-web.build-test-results-tab&runId=6247388&resultId=100105)|dotnet/runtime#87260|
|[305791](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305791)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305791&view=ms.vss-test-web.build-test-results-tab&runId=6245244&resultId=100105)|dotnet/runtime#84191|
|[306947](https://dev.azure.com/dnceng-public/public/_build/results?buildId=306947)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=306947&view=ms.vss-test-web.build-test-results-tab&runId=6244896&resultId=100105)|dotnet/runtime#87555|
|[306922](https://dev.azure.com/dnceng-public/public/_build/results?buildId=306922)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=306922&view=ms.vss-test-web.build-test-results-tab&runId=6244592&resultId=100105)|dotnet/runtime#87522|
|[306721](https://dev.azure.com/dnceng-public/public/_build/results?buildId=306721)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=306721&view=ms.vss-test-web.build-test-results-tab&runId=6238154&resultId=100105)|dotnet/runtime#87260|
|[306714](https://dev.azure.com/dnceng-public/public/_build/results?buildId=306714)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=306714&view=ms.vss-test-web.build-test-results-tab&runId=6238132&resultId=100105)|dotnet/runtime#87549|
|[306691](https://dev.azure.com/dnceng-public/public/_build/results?buildId=306691)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=306691&view=ms.vss-test-web.build-test-results-tab&runId=6238030&resultId=100105)|dotnet/runtime#86895|
|[306673](https://dev.azure.com/dnceng-public/public/_build/results?buildId=306673)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=306673&view=ms.vss-test-web.build-test-results-tab&runId=6237340&resultId=100105)|dotnet/runtime#87170|
|[306616](https://dev.azure.com/dnceng-public/public/_build/results?buildId=306616)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=306616&view=ms.vss-test-web.build-test-results-tab&runId=6235242&resultId=100105)||
|[305840](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305840)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305840&view=ms.vss-test-web.build-test-results-tab&runId=6233598&resultId=100105)|dotnet/runtime#87403|
|[306439](https://dev.azure.com/dnceng-public/public/_build/results?buildId=306439)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=306439&view=ms.vss-test-web.build-test-results-tab&runId=6231656&resultId=100105)|dotnet/runtime#87540|
|[306186](https://dev.azure.com/dnceng-public/public/_build/results?buildId=306186)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=306186&view=ms.vss-test-web.build-test-results-tab&runId=6225568&resultId=100105)|dotnet/runtime#87065|
|[305747](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305747)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305747&view=ms.vss-test-web.build-test-results-tab&runId=6214382&resultId=100105)|dotnet/runtime#87507|
|[305305](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305305)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305305&view=ms.vss-test-web.build-test-results-tab&runId=6213546&resultId=100105)|dotnet/runtime#86546|
|[305683](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305683)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305683&view=ms.vss-test-web.build-test-results-tab&runId=6212710&resultId=100105)|dotnet/runtime#87503|
|[305573](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305573)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305573&view=ms.vss-test-web.build-test-results-tab&runId=6207840&resultId=100105)|dotnet/runtime#87497|
|[305551](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305551)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305551&view=ms.vss-test-web.build-test-results-tab&runId=6207574&resultId=100105)|dotnet/runtime#87495|
|[305495](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305495)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305495&view=ms.vss-test-web.build-test-results-tab&runId=6206328&resultId=100105)|dotnet/runtime#87436|
|[305393](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305393)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305393&view=ms.vss-test-web.build-test-results-tab&runId=6204334&resultId=100105)|dotnet/runtime#87260|
|[305088](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305088)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305088&view=ms.vss-test-web.build-test-results-tab&runId=6202278&resultId=100105)|dotnet/runtime#87473|
|[305323](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305323)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305323&view=ms.vss-test-web.build-test-results-tab&runId=6202266&resultId=100105)|dotnet/runtime#87403|
|[305194](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305194)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305194&view=ms.vss-test-web.build-test-results-tab&runId=6201178&resultId=100105)|dotnet/runtime#86255|
|[305259](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305259)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305259&view=ms.vss-test-web.build-test-results-tab&runId=6201336&resultId=100105)|dotnet/runtime#87375|
|[305054](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305054)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=305054&view=ms.vss-test-web.build-test-results-tab&runId=6196514&resultId=100105)|dotnet/runtime#87403|
|[304987](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304987)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304987&view=ms.vss-test-web.build-test-results-tab&runId=6194776&resultId=100105)|dotnet/runtime#87274|
|[304949](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304949)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304949&view=ms.vss-test-web.build-test-results-tab&runId=6193566&resultId=100105)|dotnet/runtime#87469|
|[304945](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304945)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304945&view=ms.vss-test-web.build-test-results-tab&runId=6193388&resultId=100105)|dotnet/runtime#87170|
|[304881](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304881)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304881&view=ms.vss-test-web.build-test-results-tab&runId=6191874&resultId=100105)||
|[303588](https://dev.azure.com/dnceng-public/public/_build/results?buildId=303588)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=303588&view=ms.vss-test-web.build-test-results-tab&runId=6182648&resultId=100105)|dotnet/runtime#87375|
|[304493](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304493)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304493&view=ms.vss-test-web.build-test-results-tab&runId=6181830&resultId=100105)|dotnet/runtime#87436|
|[304333](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304333)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304333&view=ms.vss-test-web.build-test-results-tab&runId=6177076&resultId=100105)|dotnet/runtime#74623|
|[304226](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304226)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304226&view=ms.vss-test-web.build-test-results-tab&runId=6174470&resultId=100105)|dotnet/runtime#87425|
|[304201](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304201)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304201&view=ms.vss-test-web.build-test-results-tab&runId=6173364&resultId=100105)||
|[304188](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304188)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304188&view=ms.vss-test-web.build-test-results-tab&runId=6173352&resultId=100105)|dotnet/runtime#86722|
|[304179](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304179)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304179&view=ms.vss-test-web.build-test-results-tab&runId=6173256&resultId=100105)|dotnet/runtime#87274|
|[304175](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304175)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304175&view=ms.vss-test-web.build-test-results-tab&runId=6173240&resultId=100105)|dotnet/runtime#87326|
|[304019](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304019)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304019&view=ms.vss-test-web.build-test-results-tab&runId=6170252&resultId=100071)|dotnet/runtime#86255|
|[304112](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304112)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304112&view=ms.vss-test-web.build-test-results-tab&runId=6171870&resultId=100105)|dotnet/runtime#87065|
|[304096](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304096)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=304096&view=ms.vss-test-web.build-test-results-tab&runId=6170968&resultId=100105)|dotnet/runtime#87422|
|[301835](https://dev.azure.com/dnceng-public/public/_build/results?buildId=301835)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=301835&view=ms.vss-test-web.build-test-results-tab&runId=6168430&resultId=100105)|dotnet/runtime#86875|
|[303868](https://dev.azure.com/dnceng-public/public/_build/results?buildId=303868)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=303868&view=ms.vss-test-web.build-test-results-tab&runId=6166388&resultId=100105)|dotnet/runtime#84229|
|[303807](https://dev.azure.com/dnceng-public/public/_build/results?buildId=303807)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=303807&view=ms.vss-test-web.build-test-results-tab&runId=6165152&resultId=100105)|dotnet/runtime#87406|
|[303799](https://dev.azure.com/dnceng-public/public/_build/results?buildId=303799)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=303799&view=ms.vss-test-web.build-test-results-tab&runId=6165064&resultId=100105)|dotnet/runtime#86759|
|[303474](https://dev.azure.com/dnceng-public/public/_build/results?buildId=303474)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=303474&view=ms.vss-test-web.build-test-results-tab&runId=6164340&resultId=100105)|dotnet/runtime#87124|
|[303734](https://dev.azure.com/dnceng-public/public/_build/results?buildId=303734)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=303734&view=ms.vss-test-web.build-test-results-tab&runId=6162924&resultId=100105)|dotnet/runtime#87403|
|[301956](https://dev.azure.com/dnceng-public/public/_build/results?buildId=301956)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=301956&view=ms.vss-test-web.build-test-results-tab&runId=6163602&resultId=100105)|dotnet/runtime#87156|
|[303566](https://dev.azure.com/dnceng-public/public/_build/results?buildId=303566)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=303566&view=ms.vss-test-web.build-test-results-tab&runId=6160260&resultId=100105)|dotnet/runtime#86546|
|[303546](https://dev.azure.com/dnceng-public/public/_build/results?buildId=303546)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=303546&view=ms.vss-test-web.build-test-results-tab&runId=6159774&resultId=100105)|dotnet/runtime#86759|
|[303335](https://dev.azure.com/dnceng-public/public/_build/results?buildId=303335)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=303335&view=ms.vss-test-web.build-test-results-tab&runId=6154888&resultId=100105)|dotnet/runtime#87308|
|[303285](https://dev.azure.com/dnceng-public/public/_build/results?buildId=303285)|dotnet/runtime|[Workloads-Wasm.Build.Tests.WasmTemplateTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=303285&view=ms.vss-test-web.build-test-results-tab&runId=6151678&resultId=100105)||
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|22|60|60|

### PR DESCRIPTION
**Pull request by radical**
`_WasmDevel=true` causes the optimization flags to be `-O0`.

- This *was* automatically set whenever `Configuration=Debug`, and
`WasmBuildNative=true`, which is useful when relinking in inner loop.
- But when publishing, and relinking/AOT, `-O0` results in larger output.
- Instead, this is changed to never use `_WasmDevel` when publishing.

### [wasm] host: Add `--no-silent` command line parameter

- this is the equivalent to `--verbose`
- The default is "silent" behavior

Cannot use `--verbose` because that conflicts with `dotnet run`.

### Also:
- CI: Trigger only wasm jobs on workload-testing.targets changes, since only wasm is using this currently
- Add missing Wasm.Build.Tests.Blazor.AppsettingsTests to list of tests to run

Fixes https://github.com/dotnet/runtime/issues/87414 - side effect of the `_WasmDevel` change.

### CODE REVIEW

**Comment by radical** [comment]
The `_WasmDevel` change also fixes https://github.com/dotnet/runtime/issues/87414 .

File: src/mono/wasm/host/JSEngineHost.cs [hunk 1]
```diff
@@ -88,8 +88,11 @@ private async Task<int> RunAsync()
         int exitCode = await Utils.TryRunProcess(psi,
                                     _logger,
                                     msg => { if (msg != null) _logger.LogInformation(msg); },
-                                    msg => { if (msg != null) _logger.LogInformation(msg); });
+                                    msg => { if (msg != null) _logger.LogInformation(msg); },
+                                    silent: !_args.CommonConfig.Silent);
 
+        if (_args.CommonConfig.Silent)
+            Console.WriteLine ($"{_args.Host} exited with {exitCode}");
```
Comment (radical) [inline src/mono/wasm/host/JSEngineHost.cs:95]:
```suggestion
            Console.WriteLine($"{_args.Host} exited with {exitCode}");
```

**Comment by pavelsavara** [comment]
perhaps `--verbose-publish` or `--verbose-build` or `--verbose-aot` or `--verbose-compilation`

**Review by ilonatommy — APPROVED** [review]

File: src/mono/wasm/host/JSEngineHost.cs [hunk 2]
```diff
@@ -88,8 +88,11 @@ private async Task<int> RunAsync()
         int exitCode = await Utils.TryRunProcess(psi,
                                     _logger,
                                     msg => { if (msg != null) _logger.LogInformation(msg); },
-                                    msg => { if (msg != null) _logger.LogInformation(msg); });
+                                    msg => { if (msg != null) _logger.LogInformation(msg); },
+                                    silent: !_args.CommonConfig.Silent);
 
+        if (_args.CommonConfig.Silent)
+            Console.WriteLine($"{_args.Host} exited with {exitCode}");
```
[the remark this answers is not in the document]
Comment (radical) [reply src/mono/wasm/host/JSEngineHost.cs:95]: ugh.. good catch. I had the property with an inverted name, but before opening the PR I changed the name, but missed inverting the uses!!

**Comment by radical** [comment]
> perhaps `--verbose-publish` or `--verbose-build` or `--verbose-aot` or `--verbose-compilation`

This is for `dotnet run`, so none of these would be relevant. I don't particularly like `--no-silent` though, but don't have a good alternative right now. OTOH, `--no-*` does have precedence like `--no-build, --no-restore` etc for `dotnet run`.

File: src/mono/wasm/host/JSEngineHost.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (radical) [reply src/mono/wasm/host/JSEngineHost.cs:95]: Fixed.

**Comment by pavelsavara** [comment]
> This is for `dotnet run`, so none of these would be relevant. 

Could you please explain what are we making verbose ?

**Comment by radical** [comment]
> > This is for `dotnet run`, so none of these would be relevant.
> 
> Could you please explain what are we making verbose ?

Not much right now. The js engine exit code for example, will be shown in verbose, and the js engine invocation command line.

**Comment by radical** [comment]
@pavelsavara I'm merging this to get CI green. If we want to change the parameter name, then I can handle that in a follow up PR.

**Pull request merged**